integration-tests/api/__tests__/admin/order-edit.js

```diff
@@ -612,6 +615,233 @@ describe("[MEDUSA_FF_ORDER_EDITING] /admin/order-edits", () => {
+        variants: [
+          {
+            id: toBeAddedVariantId,
+            prices: [{ currency: "usd", amount: 200 }],
+          },
+        ],
+      })
+
+      const order = await simpleOrderFactory(dbConnection, {
+        id: orderId1,
+        fulfillment_status: "fulfilled",
+        payment_status: "captured",
+        region: {
+          id: "test-region",
+          name: "Test region",
+          tax_rate: 12.5,
+        },
+        line_items: [
+          {
+            id: lineItemId1,
+            variant_id: product1.variants[0].id,
+            quantity: 1,
+            fulfilled_quantity: 1,
+            shipped_quantity: 1,
+            unit_price: 1000,
+          },
+        ],
+      })
+
+      await simpleOrderEditFactory(dbConnection, {
+        id: orderEditId,
+        order_id: order.id,
+        created_by: "admin_user",
+      })
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      return await db.teardown()
+    })
+
+    it("creates line item that will be added to the order", async () => {
+      const api = useApi()
+
+      const response = await api.post(
+        `/admin/order-edits/${orderEditId}/items`,
+        { variant_id: toBeAddedVariantId, quantity: 2 },
+        adminHeaders
+      )
+
+      expect(response.status).toEqual(200)
+      expect(response.data.order_edit).toEqual(
+        expect.objectContaining({
+          id: orderEditId,
+          created_by: "admin_user",
+          requested_by: null,
+          canceled_by: null,
+          confirmed_by: null,
+          // "Add item" change has been created
+          changes: [
+            expect.objectContaining({
+              type: "item_add",
+              order_edit_id: orderEditId,
+              original_line_item_id: null,
+              line_item_id: expect.any(String),
+            }),
+          ],
+          /*
+           * Computed items are appended to the response
+           */
+          items: expect.arrayContaining([
+            // already existing item on the order
+            expect.objectContaining({
+              id: lineItemId1,
+              order_id: orderId1,
+              tax_lines: [],
+            }),
+            // NEW line item
+            expect.objectContaining({
+              variant: expect.objectContaining({ id: toBeAddedVariantId }),
+              quantity: 2,
+              order_id: null, // <-- NOT associated with the order at this point
+              tax_lines: [
+                expect.objectContaining({
+                  rate: 12.5,
+                  name: "default",
+                  code: "default",
+                }),
+              ],
+            }),
+          ]),
+          /*
+           * Computed totals are appended to the response
+           */
+          discount_total: 0,
+          gift_card_total: 0,
+          gift_card_tax_total: 0,
+          shipping_total: 0,
+          subtotal: 1000 + 2 * 200,
```

Comment:
question: How do you get the right amount since the tax line is associated to the new item but the item id is the original one, therefore I think that you got lucky because the allocation map contains the association of the tax line of the original item which is the same as the new one 😂 but in fact I am pretty sure it does not use the right tax lines on the new item.

Can you tell me what you think of that:
https://github.com/medusajs/medusa/pull/2246/files#diff-a26e080c56da65f0067681a51f853510451acbbe4700216d78a333826a71357dR584
https://github.com/medusajs/medusa/pull/2246/files#diff-a26e080c56da65f0067681a51f853510451acbbe4700216d78a333826a71357dR231 (this one is temporary until we agree on the way to do it)

Reply:
How did you get the right result since the tax lines are associated to the new line but in the items collection the item id is the one from the original one which is different from the one in the tax line. Therefore the totals service will not associate them during the computation.